src/Expect.elm

```diff
@@ -0,0 +1,367 @@
+
+{-| Passes if the second argument is greater than or equal to the first.
+
+    Expect.atLeast -2 (List.length [])
+
+    -- Passes because (0 >= -2) is True
+
+Failures resemble code written in pipeline style, so you can tell
+which argument is which:
+
+    -- Fails because (0 >= 3) is False
+    List.length []
+        |> Expect.atLeast 3
+
+    {-
+
+    0
+    ╷
+    │ Expect.atLeast
+    ╵
+    3
+
+    -}
+-}
+atLeast : comparable -> comparable -> Expectation
+atLeast =
+    compareWith "Expect.atLeast" (>=)
+
+
+{-| Passes if the argument is 'True', and otherwise fails with the given message.
+
+    Expect.true "Expected the list to be empty." (List.isEmpty [])
+
+    -- Passes because (List.isEmpty []) is True
+
+Failures resemble code written in pipeline style, so you can tell
+which argument is which:
+
+    -- Fails because List.isEmpty returns False, but we expect True.
+    List.isEmpty [ 42 ]
+        |> Expect.true "Expected the list to be empty."
+
+    {-
+
+    Expected the list to be empty.
+
+    -}
+-}
+true : String -> Bool -> Expectation
+true message bool =
+    if bool then
+        pass
+    else
+        fail message
+
+
+{-| Passes if the argument is 'False', and otherwise fails with the given message.
+
+    Expect.false "Expected the list not to be empty." (List.isEmpty [ 42 ])
+
+    -- Passes because (List.isEmpty [ 42 ]) is False
+
+Failures resemble code written in pipeline style, so you can tell
+which argument is which:
+
+    -- Fails because (List.isEmpty []) is True
+    List.isEmpty []
+        |> Expect.false "Expected the list not to be empty."
+
+    {-
+
+    Expected the list not to be empty.
+
+    -}
+-}
+false : String -> Bool -> Expectation
+false message bool =
+    if bool then
+        fail message
+    else
+        pass
+
+
+{-| Always passes.
+
+    import Json.Decode exposing (decodeString, int)
+    import Test exposing (test)
+    import Expect
+
+
+    test "Json.Decode.int can decode the number 42." <|
+        \() ->
+            case decodeString int "42" of
+                Ok _ ->
+                    Expect.pass
+
+                Err err ->
+                    Expect.fail err
+-}
+pass : Expectation
+pass =
+    Test.Expectation.Pass
+
+
+{-| Fails with the given message.
+
+    import Json.Decode exposing (decodeString, int)
+    import Test exposing (test)
+    import Expect
+
+
+    test "Json.Decode.int can decode the number 42." <|
+        \() ->
+            case decodeString int "42" of
+                Ok _ ->
+                    Expect.pass
+
+                Err err ->
+                    Expect.fail err
+-}
+fail : String -> Expectation
+fail =
+    Test.Expectation.Fail ""
+
+
+{-| Return `Nothing` if the given [`Expectation`](#Expectation) is a [`pass`](#pass).
+
+If it is a [`fail`](#fail), return a record containing the failure message,
+along with the given inputs if it was a fuzz test. (If no inputs were involved,
+the record's `given` field will be `""`).
```

Comment:
It would, but the overhead associated with storing that for every single test didn't seem worth the slight semantic distinction.

Currently there's no way to get a `""` value for `given` other than running a non-fuzz test, so I thought about translating that to `Nothing` and anything else to `Just`, but then it seemed like I'd be setting up a false expectation if it ended up being possible in the future for `given` to be `""` for a fuzz test.

tl;dr I agree, but the juice doesn't seem worth the squeeze. 😄